### PR TITLE
hibernate-06

### PR DESCRIPTION
**Pull request by AndreiYazadzhi**

### CODE REVIEW

File: src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -4,20 +4,24 @@
...
 import com.hibernate.service.AuthenticationService;
+import com.hibernate.service.ShoppingCardService;
 import com.hibernate.service.UserService;
 import com.hibernate.util.HashUtil;
 import java.util.Optional;
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
     private UserService userService = new UserServiceImpl();
+    private ShoppingCardService shoppingCardService = new ShoppingCardServiceImpl();
```
Comment (mfedenko09) [inline src/main/java/com/hibernate/service/impl/AuthenticationServiceImpl.java]:
```suggestion
    private UserService userService = new UserServiceImpl();
    private ShoppingCardService shoppingCardService = new ShoppingCardServiceImpl();
```
we should use @Inject here

**Review by Fedich09 — CHANGES_REQUESTED** [review]

**Review by Fedich09 — APPROVED** [review]

File: src/main/java/com/hibernate/Main.java [hunk 2]
```diff
@@ -1,21 +1,35 @@
...
+import com.hibernate.service.MovieService;
+import com.hibernate.service.MovieSessionService;
+import com.hibernate.service.ShoppingCardService;
 
 public class Main {
     private static Injector injector = Injector.getInstance("com.hibernate");
     private static final AuthenticationService authService =
             (AuthenticationService) injector.getInstance(AuthenticationService.class);
+    private static final ShoppingCardService shoppingCardService =
+            (ShoppingCardService) injector.getInstance(ShoppingCardService.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/Main.java]:
```suggestion
    private static final ShoppingCartService shoppingCartService =
            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
```

File: src/main/java/com/hibernate/dao/ShoppingCardDao.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.hibernate.dao;
+
+import com.hibernate.model.ShoppingCard;
+import com.hibernate.model.User;
+
+public interface ShoppingCardDao {
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/dao/ShoppingCardDao.java]:
```suggestion
public interface ShoppingCartDao {
```
save everywhere - change `d` to `t` please

File: src/main/java/com/hibernate/dao/impl/ShoppingCardDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.update(shoppingCard);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessException("Can`t insert shopping Card  " + shoppingCard, e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/dao/impl/ShoppingCardDaoImpl.java]:
```suggestion
            throw new DataProcessException("Can`t update shopping cart  " + shoppingCart, e);
```

File: src/main/java/com/hibernate/model/ShoppingCard.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.MapsId;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shopping_cards")
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/ShoppingCard.java]:
```suggestion
@Table(name = "shopping_carts")
```

File: src/main/java/com/hibernate/model/ShoppingCard.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+public class ShoppingCard {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/model/ShoppingCard.java]: do you need this?

File: src/main/java/com/hibernate/service/impl/ShoppingCardServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ShoppingCard shoppingCartByUser = shoppingCardDao.getByUser(user);
+        List<Ticket> tickets = shoppingCartByUser.getTickets();
+        ticketDao.add(ticket);
+        shoppingCardDao.update(shoppingCartByUser);
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/service/impl/ShoppingCardServiceImpl.java]: is `carts_tickets` table updated? what is missing here?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/hibernate/Main.java [hunk 8]
```diff
@@ -1,21 +1,35 @@
...
-import com.hibernate.util.HashUtil;
+import com.hibernate.service.MovieService;
+import com.hibernate.service.MovieSessionService;
+import com.hibernate.service.ShoppingCartService;
 
 public class Main {
     private static Injector injector = Injector.getInstance("com.hibernate");
     private static final AuthenticationService authService =
             (AuthenticationService) injector.getInstance(AuthenticationService.class);
+    private static final ShoppingCartService SHOPPING_CART_SERVICE =
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/Main.java]: let's use lowerCamelCase here, same as you do for other services

File: src/main/java/com/hibernate/Main.java [hunk 9]
```diff
@@ -1,21 +1,35 @@
...
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private static final MovieSessionService movieSessionService =
+            (MovieSessionService) injector.getInstance(MovieSessionService.class);
+    private static final MovieService movieService =
+            (MovieService) injector.getInstance(MovieService.class);
+    private static final User user = authService.register("123", "123");
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/Main.java]: let's make this a local variable

File: src/main/java/com/hibernate/dao/impl/ShoppingCartDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessException("Can`t insert shopping Card  " + shoppingCart, e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/dao/impl/ShoppingCartDaoImpl.java]:
```suggestion
            throw new DataProcessException("Can`t insert shopping Cart  " + shoppingCart, e);
```

File: src/main/java/com/hibernate/dao/impl/ShoppingCartDaoImpl.java [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    public ShoppingCart getByUser(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from ShoppingCart sc "
+                    + "left join fetch sc.tickets "
+                    + "where sc.user = :user", ShoppingCart.class)
+                    .setParameter("user", user)
+                    .getSingleResult();
+        } catch (Exception e) {
+            throw new DataProcessException("Could not find card by user: " + user, e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/hibernate/dao/impl/ShoppingCartDaoImpl.java]:
```suggestion
            throw new DataProcessException("Could not find cart by user: " + user, e);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/hibernate/Main.java (hunk 9 above)
Comment (AndreiYazadzhi) [reply src/main/java/com/hibernate/Main.java]: Checkstyle dropped error when user was local variable

Comment (AndreiYazadzhi) [reply src/main/java/com/hibernate/Main.java]: image

Comment (KseniiaMakarova) [reply src/main/java/com/hibernate/Main.java]: let's create it closer to its first usage then?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
not all comments were fixed - I also replied above

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
please make commit names informative and be attentive to comments and replies

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**